packages/data-stores/src/woocommerce-eligibility/resolvers.ts

```diff
@@ -0,0 +1,20 @@
+import { dispatch } from '@wordpress/data';
+import { wpcomRequest } from '../wpcom-request-controls';
+import { STORE_KEY } from './constants';
+import { TransferEligibility } from './types';
+
+/**
+ * Requests the site's transfer eligibility from the WPCOM API.
+ *
+ * @param   {number} siteId The id of the site to retrieve
+ */
+export function* getAutomatedTransferEligibility( siteId: number ) {
```

Comment:
Generators and controls are deprecated in latest `@wordpress/data`. Let's use thunks here:
```js
import wpcomRequest from 'wpcom-proxy-request';

const getAutomatedTransferEligibility = ( siteId ) => async ( { dispatch } ) => {
  dispatch.requestTransferEligibility( siteId );
  const transferEligibility = await wpcomRequest( { path, apiVersion } );
  dispatch.receiveTransferEligibility( transferEligibility, siteId );
}
```

Reply:
@alshakero All of the existing data stores are using generators.

Do you have any thoughts on switching to thunks?

Reply:
Like, they are not formally deprecated in a "wrapped with `@wordpress/deprecated` and triggering a warning" sense, but Gutenberg itself has been fully migrated away and we should migrate plugins, too, or at least, don't introduce new usages.

Reply:
@jsnajdr Yeah, if thunks are the direction we're going, we should definitely make sure new code uses them.

Can you offer any advice on how to handle typing that `dispatch` object? I'm struggling a bit with that.

Reply:
It seems that Calypso is running version `6.1.5` of the `@wordpress/data` package. In this versions thunks don't seem to be enabled. I will be doing some more testing but it feels like we wont be able to go this way unless we update the `@wordpress/data` package.

Reply:
In that version, you need to enable them with a flag:
```js
registerStore( name, {
  reducer,
  actions,
  selectors,
  __experimentalUseThunks: true
} );
```
Only the latest version of `@wordpress/data`, shipped in WordPress 6.0, enables them by default.